packages/access-api/test/provider-add.test.js

```diff
@@ -97,6 +99,100 @@ for (const accessApiVariant of /** @type {const} */ ([
       })
     })
   })
+
+  if (
+    ['provider/add', 'access/delegate'].every((f) =>
+      accessApiVariant.features.has(f)
+    )
+  ) {
+    it('provider/add allows for access/delegate', async () => {
+      const space = await principal.ed25519.generate()
+      const issuer = await accessApiVariant.issuer
+      const service = await accessApiVariant.audience
+      const accountDid = /** @type {const} */ ('did:mailto:example.com:foo')
+      const serviceSessionAttest = await ucanto.delegate({
+        issuer: service,
+        audience: issuer,
+        capabilities: [
+          {
+            can: 'provider/add',
+            with: accountDid,
+          },
+          {
+            can: 'access/delegate',
+            with: space.did(),
+          },
+        ],
+      })
+      const sessionProofs = [
+        await ucanto.delegate({
+          issuer: service,
+          audience: issuer,
+          capabilities: [
+            {
+              can: 'ucan/attest',
+              with: accountDid,
+              nb: {
+                // note: whole delegation is also included in 'proofs'
+                proof: serviceSessionAttest.cid,
+              },
+            },
+          ],
+        }),
+        serviceSessionAttest,
+      ]
+      const addStorageProvider = await ucanto
+        .invoke({
+          issuer,
+          audience: service,
+          capability: {
+            can: 'provider/add',
+            with: accountDid,
+            nb: {
+              provider: 'did:web:web3.storage:providers:w3up-alpha',
+              consumer: space.did(),
+            },
+          },
+          proofs: [
+            // space says issuer can provider/add with this account
+            await ucanto.delegate({
+              issuer: space,
+              audience: issuer,
+              capabilities: [
+                {
+                  can: 'provider/add',
+                  with: accountDid,
+                },
+              ],
+            }),
+            ...sessionProofs,
+          ],
+        })
+        .delegate()
+      const addStorageProviderResult = await accessApiVariant.invoke(
+        addStorageProvider
+      )
+      assertNotError(addStorageProviderResult)
```

Comment:
I suspect this is another instance where validator is not used, if that is not the case I would be very concerned. And if validator is disabled indeed, perhaps it's better to enable it so we don't end up in situation where tests pass but deployed code does not.

Reply:
I will check that, but I don't think so.
`accessApiVariant` looks to be built by the `createTesterFromContext` that tests against miniflare. And I traced that to a path that does `Server.provider(Provider.add, ...`. I will check outside of github though

Reply:
We met in person, and I think we verfied that the validator is being used.